org-cyf-piscine/content/piscine/projects/music-data/index.md

```diff
@@ -0,0 +1,72 @@
+
+TODO: Generate and link to `data.js`
+
+`data.js` is a file containing three functions:
+1. `getUserIDs()`: when called, returns an array of strings, each of which is a user ID.
+1. `getListens(userID)`: when called, returns an array of objects, each of which contains information about a single time that the given user listened to a song. The listens are sorted by when they happened, oldest to newest.
+2. `getSong(songID)`: when called with one string as an argument, returns an an object containing information about a single song.
+
+## Requirements
+
+You must submit both a link to your GitHub repo, and a link to the deployed website.
+
+Your website must be hosted on the internet, and must be automatically deployed when you merge changes to your GitHub repo.
+
+Your website must include a drop-down to select a user to display information for. When a user is selected, you must display answers to all of the questions for that user. If a question doesn't apply to that user, you should not show the question.
+
+You **must not** implement any kind of authentication. Just a drop-down to choose which user's information to display.
+
+Your GitHub repository must contain unit tests which demonstrate that your code works. End to end tests are optional.
+
+## Questions to answer
+
+1. What was the user's most often listened to song according to the data?
+2. What was the user's most often listened to artist according to the data?
+3. What was the user's most often listened to song on Friday nights (between 5pm and 4am)?
+4. If we were talking about _listening time_ rather than _number of listens_, would any of the above questions have different answers?
+5. What song did the user listen to the most times in a row (i.e. without any other song being listened to in between)? How many times was it listened to?
+6. Are there any songs that the user listened to every day?
+7. What were the user's top three genres to listen to?
+
+## Rubric
+
+TODO: Unclear whether we want this to be in the trainee-facing part of the project, or hidden, or to have the same bullet points in both but more information only available to assessors?
+
+All of the below requirements must be met for the project to be considered complete:
+
+* The website must contain a drop-down which lists three users.
+* Selecting a user must display answers relevant to that user (see table below).
+* The code written to calculate the answers to the questions must seem like it could handle different data if it were supplied, including the following edge-cases:
+  * An empty array of song data, or an empty subset for any question, should completely hide the question and answer. Displaying the question and an empty result, or any kind of error, is not acceptable. If no questions apply to the user, some intelligible statement should be shown to the user (e.g. "This user didn't listen to any songs.").
```

Comment:
Given that the relevant instruction above is phrased as "If a question doesn't apply to that user, you should not show the question" implies that the question doesn't make sense, not that there isn't available data.

In other words, a reasonable answer to "Are there any songs that the user listened to every day?" is "No" - so should the answer be displayed or not here?

Reply:
Added the text: "If the answer is yes, you should show which one(s). If the answer is no, you should not show anything about this question."